### ISSUE TITLE
Possible bug in latest dev Theano install with nvidia driver 378.13

### ISSUE THREAD

**Issue by mpgussert** [Theano/Theano#5707]
Greetings all!

I don't know if this is coming from Theano or Keras, but I will be posting this on their Issues page as well.

Forcing driver 378.13 for Theano causes softmax not to normalize on the GPU.  Here's the code

```
import numpy as np
import keras
from keras.models import Sequential, Model
from keras.layers import Input, Dense, Convolution2D, MaxPooling2D, UpSampling2D, Cropping2D, Flatten
from keras.regularizers import l2, l1, l1l2
from keras import backend as K

def atariPolicyModel(state, actions):
  model = Sequential()
  model.add(Convolution2D(16,8,8,subsample=(4,4), input_shape=state,  border_mode='same', activation='relu'))
  model.add(Convolution2D(32,4,4,subsample=(2,2), border_mode='same', activation='relu'))
  model.add(Flatten())
  model.add(Dense(256, activation='relu'))
  model.add(Dense(actions, activation='softmax'))
  return model
  
state = (4,210,160)
actions = 8

In = np.random.rand(1,4,210,160)

model1 = atariPolicyModel(state, actions)

print np.sum(model1.predict(In))

```

In Ubuntu 16.04...

```
mpgussert@Cthulhu:KerasAC$ THEANO_FLAGS=device=cuda0 ipython -i Tests.py 
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
Type "copyright", "credits" or "license" for more information.

IPython 5.1.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.
Using Theano backend.
WARNING: loading blacklisted driver because the load was forced.
Using cuDNN version 5105 on context None
Mapped name None to device cuda0: Quadro GP100 (0000:02:00.0)
1.63243

In [1]: exit
mpgussert@Cthulhu:KerasAC$ ipython -i Tests.py 
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
Type "copyright", "credits" or "license" for more information.

IPython 5.1.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.
Using Theano backend.
1.0

In [1]: exit

```

can anyone else verify this behavior?

Thanks!

**Comment by abergeron** [comment]
As was made evident by the message you bypassed, there was a problem with Theano and that version of the driver.

Please update both Theano and libgpuarray to the most recent version and try again.

**Comment by mpgussert** [comment]
I am currently away from my computer.  I will update tomorrow, try again, and post the results here

**Comment by mpgussert** [comment]
I have updated libgpuarray and theano.  Current versions are

Theano:  0.9.0rc1.dev-RELEASE
libgpuarray: v0.6.2

the problem persists.

```
mpgussert@Cthulhu:KerasAC$ THEANO_FLAGS=device=cuda0 ipython -i Tests.py 
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
Type "copyright", "credits" or "license" for more information.

IPython 5.1.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.
Using Theano backend.
Using cuDNN version 5105 on context None
Mapped name None to device cuda0: Quadro GP100 (0000:02:00.0)
1.62243

In [1]: exit
mpgussert@Cthulhu:KerasAC$ 
```

Are these not the newest version?  if not, can you point me to them?
Thanks!

**Comment by nouiz** [comment]
That isn't the latest Theano version. UPdate Theano to the dev version:

http://deeplearning.net/software/theano_versions/dev/install_ubuntu.html#bleeding-edge-installation-recommended

pip install --user --no-deps git+https://github.com/Theano/Theano.git#egg=Theano

**Comment by mpgussert** [comment]
```
mpgussert@Cthulhu:KerasAC$ pip install --no-deps git+https://github.com/Theano/Theano.git#egg=Theano
Requirement already satisfied: Theano from git+https://github.com/Theano/Theano.git#egg=Theano in /usr/local/lib/python2.7/dist-packages
mpgussert@Cthulhu:KerasAC$ ipython 
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
Type "copyright", "credits" or "license" for more information.

IPython 5.1.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: import theano

In [2]: theano.__version__
Out[2]: '0.9.0rc1.dev-RELEASE'

In [3]: exit
mpgussert@Cthulhu:KerasAC$ 
```
that's the dev version i get from pip

**Comment by nouiz** [comment]
Sorry try:

pip install --upgrade --no-deps
git+https://github.com/Theano/Theano.git#egg=Theano

**Comment by mpgussert** [comment]
GLORIOUS!

```
mpgussert@Cthulhu:KerasAC$ THEANO_FLAGS=device=cuda0 ipython -i Tests.py 
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
Type "copyright", "credits" or "license" for more information.

IPython 5.1.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.
Using Theano backend.
Using cuDNN version 5105 on context None
Mapped name None to device cuda0: Quadro GP100 (0000:02:00.0)
1.0

In [1]: 
```

Thank you very much!  That did the trick! =D
